### PR TITLE
chore(docs): rename killer to enforcer

### LINKED ISSUE #2517: update docs to use NotifyEnforcer
https://github.com/cilium/tetragon/blob/main/docs/content/en/docs/concepts/tracing-policy/selectors.md#notify-killer-action

see: https://github.com/cilium/tetragon/pull/2117

### PR DESCRIPTION
**Pull request by mozillazg**
Closes #2517

### Description

update docs to use NotifyEnforcer

### Changelog

```release-note
chore(docs): rename killer to enforcer
```

### CODE REVIEW

File: bpf/process/types/basic.h [hunk 1]
```diff
@@ -107,7 +107,7 @@ enum {
 	ACTION_SIGNAL = 9,
 	ACTION_TRACKSOCK = 10,
 	ACTION_UNTRACKSOCK = 11,
-	ACTION_NOTIFY_KILLER = 12,
+	ACTION_NOTIFY_ENFORCER = 12,
```
Comment (mtardy) [inline bpf/process/types/basic.h:110]: is it okay @kkourt, I think we wanted to rename but this is technically a breaking change for API users now!

**Review by mtardy — COMMENTED** [review]
Thanks for finishing this, let me just ask a few question first :)

File: bpf/process/types/basic.h (hunk 1 above)
Comment (kkourt) [reply bpf/process/types/basic.h:110]: It's  only internal, correct? So i think it should be fine.

**Comment by mtardy** [comment]
Hey! I think we can merge that, thanks for the work! Could you bump the patch version of the CRD since those changes touch it, it's just here (don't forget to run `make crds` after that):

https://github.com/cilium/tetragon/blob/88036cd6e90f875744840d4ec168d8b50fa54050/pkg/k8s/apis/cilium.io/v1alpha1/version.go#L10

And can you please squash your commits into the first one when it's done?

File: bpf/process/types/basic.h (hunk 1 above)
Comment (mtardy) [reply bpf/process/types/basic.h:110]: nevermind I confused myself I thought it was a protobuf enum, no worry then.

**Review by mtardy — APPROVED** [review]
Thanks!

**Pull request merged**